src/plugins/csv/csvplugin.cpp

```diff
@@ -43,62 +45,64 @@ CsvPlugin::CsvPlugin()
 
 bool CsvPlugin::write(const Map *map, const QString &fileName)
 {
+    // Extract file name without extension and path
+    QFileInfo fileInfo(fileName);
+    fileInfo.setCaching(false);
+    const QString fileNameWoExtension = fileInfo.baseName();
+    const QString filePath = fileInfo.path();
+
+    // Traverse all tile layers
+    foreach (const Layer *layer, map->layers()) {
+        if (layer->layerType() != Layer::TileLayerType)
+            continue;
+            
+        // Get the output file name for this layer
+        const TileLayer *tileLayer = static_cast<const TileLayer*>(layer);
+        const QString layerName = tileLayer->name();
+        const QString layerFileName = fileNameWoExtension + QString("_") + layerName + QString(".csv");
+        const QString layerFileNameWPath = QDir(filePath).filePath(layerFileName);
+            
 #ifdef HAS_QSAVEFILE_SUPPORT
-    QSaveFile file(fileName);
+        QSaveFile file(layerFileNameWPath);
 #else
-    QFile file(fileName);
+        QFile file(layerFileName);
```

Comment:
This should also use `layerFileNameWPath`. Please just name that `layerFilePath`, btw.